python/shotgun_model/shotgun_model.py

```diff
@@ -440,6 +440,7 @@ def _load_data(
                                           each column in the row contains the value for the corresponding field from columns. This means
                                           that the data from the loaded entity will be available field by field. Subclasses can modify
                                           this behavior by overriding _get_additional_columns.
+        :param editable_columns:          A subset of ``columns`` that will be editable in views that use this model.
```

Comment:
@robblau: The `columns` param allows you to specify the order of the columns to display. It's also already a subset of the `fields` param. My personal preference would be to keep it this way for consistency, but I'm not passionate about it either way. I've added a check to ensure it is a subset for now. 

@manneohrstrom: The items are created by the model and that's typically where the editable state is set for standard item models. I think the views can define edit triggers to allow disallow for a particular view. That seem reasonable?

Regarding the single, editable column, yes you can do that. I feel like there are so many options here, this may better be relayed via the test harness/example app we need to have. Thoughts?